paramiko/hostkeys.py

```diff
@@ -339,7 +340,9 @@ def from_line(cls, line, lineno=None):
         :param str line: a line from an OpenSSH known_hosts file
         """
         log = get_logger("paramiko.hostkeys")
-        fields = re.split(" |\t", line)
+        fields = re.split(" +|\t+", line, 1)
```

Comment:
Can you explain why you've implemented this in this manner?

Specific questions:

- This first split (with `maxsplit=1`) will always produce a list of length two, so why bother with the `if len(fields) == 2`? Just split twice.

- Why are you splitting on a run of spaces or a run of tabs in the first split, but only a single space or tab in the second split? Does that match the OpenSSH behavior? Either way, is there any reason not to _also_ allow a run of spaces or tabs for the second whitespace separator?

- What about cases where spaces and tabs are intermixed in the whitespace? Is that valid for OpenSSH? If not, does it matter? Would it be good for paramiko to tolerate those cases also? If so, then this isn't the right regex to use.

My inclination would be to replace these three lines with something like the following (passing `maxsplit` as a keyword arg so that it's clear what that value means):

```
fields = re.split(r"[ \t]+", line, maxsplit=2)
```

But, there might be reasons why this is a bad idea. Quite open to being told why we shouldn't do this. 😀

Reply:
The idea was to split with more than one whitespace / tab only the hostname and the key. But I actually tested it now with OpenSSH 9.0p1 and it's possible to split the keytype and base64-encoded key with more than one whitespace character.
Mixing spaces and tabs is supported too.